activerecord/lib/active_record/railtie.rb

```diff
@@ -357,30 +357,35 @@ class Railtie < Rails::Railtie # :nodoc:
     end
 
     initializer "active_record_encryption.configuration" do |app|
-      ActiveRecord::Encryption.configure \
-         primary_key: app.credentials.dig(:active_record_encryption, :primary_key),
-         deterministic_key: app.credentials.dig(:active_record_encryption, :deterministic_key),
-         key_derivation_salt: app.credentials.dig(:active_record_encryption, :key_derivation_salt),
-         **config.active_record.encryption
+      ActiveRecord::Encryption.install_encrypted_attributed_declared_hook
 
-      ActiveSupport.on_load(:active_record) do
```

Comment:
`ActiveSupport.on_load(:active_record)` should only run during eager loading phase, that is `after_initialize`, so this change isn't necessary. Why do you need to do this?

Why don't you move the `ActiveRecord::Encryption.configure` call to inside `ActiveSupport.on_load(:active_record) do`? I don't think we need to use `after_initialize` here.

Reply:
hey @rafaelfranca

thanks for the tip! We are moving this PR to https://github.com/rails/rails/pull/48530, which also fixes the initialization problem. 

about setting ActiveRecord::Encryption.configure on `ActiveSupport.on_load(:active_record)`, I found the on my app my ActiveRecord was loaded before my initializers got executed. See [this](https://github.com/rails/rails/pull/48530#issuecomment-1601076219) and [this](https://github.com/rails/rails/pull/48530#issuecomment-1601178500) for the @jorgemanrubia thoughts. 

> It it possible that some initializer in this app is loading an Active Record model? That will trigger the AR encryption initialization logic. application.rb runs before any initializer runs, so the fact that it works makes me suspect that something is triggering that initialization before the new defaults' initializer is running.

After some debugging I found that when I use https://github.com/basecamp/name_of_person gem it loads the active_record before running any initializer. See [this](https://github.com/rails/rails/pull/48530#issuecomment-1601279484)

So I think @jorgemanrubia will rely on `after_initialize` too on his new PR

Reply:
That is a bug in your application and in the [name_of_person](https://github.com/basecamp/name_of_person) gem that should be fixed. Rails should not change its architecture because of bugs in other libraries.